dataland-internal-storage/src/main/kotlin/org/dataland/datalandinternalstorage/services/DatabaseDataStore.kt

```diff
@@ -1,28 +1,73 @@
+ * @param dataItemRepository
+ * @param cloudEventMessageHandler service for managing CloudEvents messages
+ * @param nonPersistedDataClient the service for retrieving data from the temporary storage
+ * @param objectMapper object mapper used for converting data classes to strings and vice versa
  */
 @Component
 class DatabaseDataStore(
     @Autowired private var dataItemRepository: DataItemRepository,
-) {
+    @Autowired var cloudEventMessageHandler: CloudEventMessageHandler,
+    @Autowired var nonPersistedDataClient: NonPersistedDataControllerApi,
+    @Autowired var objectMapper: ObjectMapper,
+) { companion object {
+    private const val storageQueue = ("\${spring.rabbitmq.storage-queue}")
+    private const val storedQueue = ("\${spring.rabbitmq.stored-queue}")
+}
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    /**
+     * Method that listens to the storage_queue and stores data into the database in case there is a message on the
+     * storage_queue
+     * @param message Message retrieved from storage_queue
+     */
+
+    @RabbitListener(queues = [storageQueue])
+    fun listenToStorageQueueAndTransferDataFromTemporaryToPersistentStorage(message: Message) {
+        val dataId = cloudEventMessageHandler.bodyToString(message)
+        val correlationId = message.messageProperties.headers["cloudEvents:id"].toString()
+        logger.info("Received DataID $dataId and CorrelationId: $correlationId")
+        val data = nonPersistedDataClient.getCompanyAssociatedDataForInternalStorage(dataId)
+        logger.info("Received DataID $dataId and DataDataDataStoreStoreStore: $data")
+        logger.info("Inserting data into database with dataId: $dataId and correlation id: $correlationId.")
+        insertDataAndSendMessage(dataId, data, correlationId)
+    }
 
     /**
-     * Insterts data into a database
-     * @param data a json object
-     * @return id associated with the stored data
+     * Method to actually insert data into the database and send a message to stored_queue after it has finished
+     * @param dataId to identify the data as first property of item to store
+     * @param data the data to be stored in the storage
+     * @param correlationId of the request initiating the storing of data
      */
-    fun insertDataSet(data: String): String {
-        val dataID = "${UUID.randomUUID()}:${UUID.randomUUID()}_${UUID.randomUUID()}"
-        dataItemRepository.save(DataItem(dataID, data))
-        return dataID
+    @RabbitHandler
+    fun insertDataAndSendMessage(dataId: String, data: String, correlationId: String) {
+        try {
+            dataItemRepository.save(DataItem(dataId, objectMapper.writeValueAsString(data)))
+            cloudEventMessageHandler.buildCEMessageAndSendToQueue(
+                dataId, "Data successfully stored", correlationId, storedQueue,
+            )
+        } catch (exception: IllegalArgumentException) {
```

Comment:
Why should there be an IllegalArgumentException? I would expect tons of database exceptions from the save command, and tons of IO exceptions from the buildCEMessageAndSendToQueue method. But I don't see why there should ever occur an IllegalArgumentException.

I feel the error handling here doesn't add any value.

Reply:
if it's removed anyways, this method (`insertDataAndSendMessage`) can be inlined.

Reply:
Agreed. Done

Reply:
ok